Makefile.common

```diff
@@ -1,9 +1,10 @@
 # common parts of Makefile and Makefile.mingw
 
-COMPILE.c = $(CC) $(CFLAGS) $(CPPFLAGS) $(TARGET_ARCH) -c
+COMPILE.c = $(CC) $(CFLAGS) $(LDFLAGS) $(CPPFLAGS) $(TARGET_ARCH) -c
```

Comment:
why is this useful? this is the rule for turning .c into .o, which is nothing to do with linking